satellite/contact/service.go

```diff
@@ -17,24 +22,78 @@ var Error = errs.Class("contact")
 
 var mon = monkit.Package()
 
+// Config contains configurable values for contact service
+type Config struct {
+	ExternalAddress string `user:"true" help:"the public address of the node, useful for nodes behind NAT" default:""`
+}
+
+// Conn represents a connection
+type Conn struct {
+	conn   *grpc.ClientConn
+	client pb.NodesClient
+}
+
 // Service is the contact service between storage nodes and satellites.
 // It is responsible for updating general node information like address, capacity, and uptime.
 // It is also responsible for updating peer identity information for verifying signatures from that node.
 //
 // architecture: Service
 type Service struct {
-	log       *zap.Logger
+	log *zap.Logger
+
+	mutex sync.Mutex
+	self  *overlay.NodeDossier
+
 	overlay   *overlay.Service
 	peerIDs   overlay.PeerIdentities
 	transport transport.Client
 }
 
 // NewService creates a new contact service.
-func NewService(log *zap.Logger, overlay *overlay.Service, peerIDs overlay.PeerIdentities, transport transport.Client) *Service {
+func NewService(log *zap.Logger, self *overlay.NodeDossier, overlay *overlay.Service, peerIDs overlay.PeerIdentities, transport transport.Client) *Service {
 	return &Service{
 		log:       log,
+		mutex:     sync.Mutex{},
```

Comment:
this line can be removed, mutex doesn't need to be initialized. The [go docs](https://golang.org/pkg/sync/#Mutex)  say "sync.Mutex does not have an explicit constructor or Init method. The zero value for a sync.Mutex is an unlocked mutex."

Reply:
oh ok cool